packages/plasma-new-hope/src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { forwardRef, useEffect } from 'react';
+
+import type { RootProps } from '../../engines';
+import { cx } from '../../utils';
+import {
+    IconStarFill16,
+    IconStarFill24,
+    IconStarFill36,
+    IconStarHalfFill16,
+    IconStarHalfFill24,
+    IconStarHalfFill36,
+    IconStarOutline16,
+    IconStarOutline24,
+    IconStarOutline36,
+} from '../_Icon';
+
+import type { RatingProps } from './Rating.types';
+import { base, Content, HelperText, Score, Star, StarsWrapper, Wrapper } from './Rating.styles';
+import { base as viewCSS } from './variations/_view/base';
+import { base as sizeCSS } from './variations/_size/base';
+import { getHelperTextPlacement, helperTextPlacements, valueConstraints } from './utils';
+import { classes, privateTokens } from './Rating.tokens';
+
+export const ratingRoot = (Root: RootProps<HTMLDivElement, RatingProps>) =>
+    forwardRef<HTMLInputElement, RatingProps>(
+        (
+            {
+                value,
+                hasValue = true,
+                precision = 1,
+                valuePlacement,
+
+                iconSlot,
+                iconSlotOutline,
+                iconSlotHalf,
+                hasIcons,
+                iconQuantity = 5,
+
+                helperText,
+                size,
+                view,
+                ...rest
+            },
+            ref,
+        ) => {
+            let innerValue = (value && String(value)) || '';
```

Comment:
```suggestion
            let innerValue = value ? String(value)) : '';
```
так вроде бы попроще